client/js/components/addon/AddonWrapper.vue

```diff
@@ -23,6 +15,15 @@ export default {
   name: 'AddonWrapper',
 
   props: {
+    /**
+     * The addonProps prop will be bound to the addon component to add props dynamically.
+     */
+    addonProps: {
+      type: Object,
+      required: false,
+      default: () => {}
+    },
```

Comment:
even though its not new. How does this works. The core should not know anything about the addon..

Reply:
Sometimes addon components need props which and therefore those props must be set dynamically. This way we can pass down required props directly from the wrapper to the corresponding addon component. However what will be done with those props in the addon is something the core does not know before the addon is mounted

Reply:
but how do we know which props to pass...
I would like to discuss somewhere if we can find another solution wich is more flexible